pkgs/nix-tools/darwin-rebuild.sh

```diff
@@ -112,6 +116,10 @@ done
 
 if [ -z "$action" ]; then showSyntax; fi
 
+if [ "$updateChannels" = true ] ; then
+  nix-channel --update darwin
```

Comment:
Hmm, that would fail in a lot of cases, where people use nix as a deamon service and therefore root owns it.

Reply:
You're right… is it reasonable to try to catch all that cases?

Reply:
You could test if the nix-channel exists and if nix runs as a daemon.
However the latter needs sudo, to inspect the running services.

(There's also the possibility that people have nix-darwin as a user and a root channel, without knowing it. I guess that's currently a bug in the installer. At least to my understanding.)

Reply:
Oh?! I think I have the scenario described here: root channel and nixpkgs, user nix-darwin. If that's "wrong", how can I clean that up?

Reply:
Maybe that command line argument is a good occasion to throw a user warning/advice if it detects a "wrong" configuration?